### ISSUE TITLE
`edit` a constructor for a decl with no name is a round-trip error

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#2886]
Something like:
````
```unison
unique type Days = Mon | Tue | Wed | Thu | Fri
```
```ucm
.> add
.> delete.type Days
.> edit Days.Mon
.> load
```
````

with scratch.u
```
unique type #ujdhd1f2ne
  = Days.Mon
  | Days.Tues
  | Days.Wed
  | Days.Thurs
  | Days.Fri
```

**Comment by ceedubs** [comment]
@aryairani what do you think that the behavior _should_ be in this case? The behavior that you've shown seems fairly reasonable to me. But I guess that it's problematic that a unique type declaration would try to give its own hash.

**Comment by sellout** [comment]
Is this actually a round-trip bug? Shouldn’t deleting a unique type fail if there are dependencies on that type?

**Comment by ceedubs** [comment]
> Shouldn’t deleting a unique type fail if there are dependencies on that type?

Yes. But there are currently some issues such as #5044 that can lead to this situation. And there were even more in the past.
